app/controllers/discovered_hosts_controller.rb

```diff
@@ -80,6 +80,7 @@ def update
           delete_discovery_attribute_set(@host.id)
           process_success :success_redirect => host_path(@host), :redirect_xhr => request.xhr?
         else
+          @host.expire_token
```

Comment:
Trying to follow the logic here... is the problem that a token is created during the attempt to provision the host, but not deleted if it fails? If so, why does the transaction not catch that?